param_sweeps/driver.py

```diff
@@ -92,10 +92,13 @@ def parameter_sets(self) -> dict:
 class SweepDriver:
     """Sweeps parameters of a worker driver."""
 
-    def __init__(self, params):
+    def __init__(self, params: SweepParams):
         self.params: SweepParams = params
-        self.workspace = params.geoh5
-        self.working_directory = os.path.dirname(self.workspace.h5file)
+        if isinstance(params.geoh5, Workspace):
+            self.workspace = params.geoh5
+        else:
+            self.workspace = Workspace(params.geoh5, mode="r")
```

Comment:
if workspace is given obtained as a string or Path (as a consequence of `plain_data`)